interfaces/cython/cantera/test/test_mixture.py

```diff
@@ -112,10 +112,11 @@ def test_charge(self):
 
     def test_phase_moles(self):
         M = self.mix.phase_moles()
+        name = self.phase2.name
         self.assertEqual(M[0], self.mix.phase_moles(0))
-        self.assertEqual(M[1], self.mix.phase_moles('air'))
+        self.assertEqual(M[1], self.mix.phase_moles(name))
 
-        self.mix.set_phase_moles('air', 4)
+        self.mix.set_phase_moles(name, 4)
```

Comment:
The fact that this no longer works as it originally did is a clear downside of making the the name globally unique.

Reply:
The test suite creates numerous `Solution` objects, each of which creates a name entry. In most use cases, the automatic renaming won't get triggered (or a user may prefer a custom name), so I don't think that there is a huge downside here.

Reply:
I wouldn't want to presume anything about the level of complexity of a users code, and how many objects with the same name may be initialized. I'm thinking especially of interactive use in an environment like a Jupyter notebook, where the easiest thing to do is often to re-run the initialization code, in which case this will result in the names all changing.

I do not want to enforce a requirement of globally unique names, which has a non-negligible usability cost, when it is not necessary as part of the current interface or for features (like serialization) that we want to implement.

Reply:
Per the above discussion, the `phase_id` is unique, which (I believe) makes this concern moot: in most cases, the name is the same as the phase id (for the first instance), so I do not expect undesired behavior coming up often. Will change the unit test accordingly.

At the same time, identification via `name` should remain possible? (even if it is cumbersome)

Reply:
Resolved by switching to name throughout.